.github/workflows/publish.yml

```diff
@@ -0,0 +1,52 @@
+name: publish
+on:
+  push:
+    tags:
+      - 'v.*'
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Build package
+        run: |
+          python -m pip install wheel jupyter-packaging jupyterlab>=3
+          BUILD_JUPYTERLAB_EXTENSION=1 python setup.py sdist bdist_wheel
+          # Don't publish a tar.gz file over 1MB  (Issue #730)
+          if (($(wc -c < dist/*.tar.gz) > 1000000)); then exit 1; fi
```

Comment:
What's the deal with this test? It should be in the CI.

But beware that distro pacakagers rely on the fact that `sdist` is a superset of `git archive`. Of course you can remove files like `.github`, but consult with a distro packager to be safe. E.g. the maintainers of `python3-jupytext` on fedora. I can do a quick lookover if you point me to the sdist ignore list.

Reply:
My reading of that check was so that you make sure that large filea are not being packaged right? Well about that, they should be packaged because distro maintainers expect that all relevant files are available on PyPI as well as other generated files.